--- hula-0.0.234/src/agents/nmap/guid.c	2005-05-25 09:29:47.000000000 +0100
+++ hula-0.0.230/src/agents/nmap/guid.c	2005-06-08 20:06:50.213858352 +0100
@@ -87,7 +87,6 @@
     return;
 }
 #elif defined(LINUX)
-#include <sys/sysinfo.h>
 
 void 
 GuidReset(void)
@@ -97,7 +96,6 @@
     unsigned char salt[32];
     unsigned char digest[SHA_DIGEST_LENGTH];
     unsigned char *ptr;
-    struct sysinfo si;
     struct timeval tv;
     SHA_CTX context;
 
@@ -114,12 +112,8 @@
         SHA1_Update(&context, name, sizeof(name) - 1);
     }
 
-    if (!sysinfo(&si)) {
-        SHA1_Update(&context, &si, sizeof(si));
-    } else {
-        RAND_bytes(name, sizeof(name) - 1);
-        SHA1_Update(&context, name, sizeof(name) - 1);
-    }
+    RAND_bytes(name, sizeof(name) - 1);
+    SHA1_Update(&context, name, sizeof(name) - 1);
 
     SHA1_Final(digest, &context);
 
